src/tensor_ops.jl

```diff
@@ -256,6 +256,23 @@ end
 Base.ctranspose(S::AllTensors) = transpose(S)
 
 
+############################
+# Symmetric/Skew-symmetric #
+############################
+
+"""
+Computes the symmetric part of a second order tensor.
+"""
+@inline sym{dim,T}(S1::Tensor{2,dim,T}) = convert(SymmetricTensor{2,dim,T},S1)
```

Comment:
It will return a symmetric tensor. The question is if it should return a `SymmetricTensor`.

The reason why this is problematic is that if functions changes the input type too frequently it becomes impossible to track what type variables have in your head. It can also lead the performance problems which are hard to spot:

For example:

```jl
function foo()
    A = rand(Tensor{2,3})
    A = sym(A)
    return A
end
```

Using `@code_warntype` this shows:

```jl
Variables:
  #self#::#foo
  A::Any
```

which means that `A` is type unstable (the same variable changes type in a function) and will have VERY bad performance. It is therefore better in my opinion to have to be explicit when you want to convert from one type to another. Let's think about it for a while :P